### ISSUE TITLE
Engagement window SDK implementation

### ISSUE THREAD

**Issue by waijin** [LivePersonInc/developers-community#187]
Greetings, 

I'm trying to implement the [Engagement Window Widget SDK](https://developers.liveperson.com/rt-interactions-window-sdk-overview.html). However I find the guide on the website to be really vague. I've contacted customer support and they directed me here, could you provide me more information or even a step-by-step guide on how to implement this correctly ? 

Thanks.
Jin

**Comment by scottwestover** [comment]
Hi Jin,

To implement the widget, after you include the js sdk file, the main two methods you need to call are: `init` and `bind`. 

By initializing the sdk, it allows the main engagement window to pass conversation events to your window, and you choose which events to listen to by binding to them.

When you `bind` to the events, you pass the sdk a callback function that will be called any time one of these events are triggered.

Example from the docs:
```
{
  // 1 - Initialize the sdk
  var lpWidgetSDK = lpTag.LPWidgetSDK.init();
 // 2 - choose the events you would like to bind to
  lpWidgetSDK.bind(lpTag.LPWidgetSDK.API.events.CONVERSATION_INFO, _onEvent, this);

 // 3 - the function that will be called any time this event is triggered
  function _onEvent(eventData) {
    if (eventData) {
        console.log(eventData);

}
```

Here is a sample widget that was built with the SDK that will load a YouTube video if a certain message is passed during chat: https://github.com/LiveEngage-Examples/engagement-window-widget-youtube/blob/master/index.html

Basically, the widget is checking each chat line to see if it contains the text `See this video:`, and if it does, it will grab the YouTube link and load it in an Iframe. Example chat line: `See this video: https://www.youtube.com/watch?v=5VKLUcx5VKg`

**Comment by scottwestover** [comment]
Also, to enable the engagement window widget on your account, someone from your account team will need to enable the widget on your account.

If you have any more questions, just let us know.

**Comment by timu** [comment]
Hi @scottwestover. your example helped me much so thanks for sharing that.
I only have one question my mind. How is the actual widget window can be triggered to show on specific events and close when it's done? 

If I understand correctly your widget example shown all the time when the chat is active and it shows youtube videos if it catches the certain phrase. What I wonder is how widget window can be closed at the start of the conversation and can be opened only if the certain phrase written in the chat.

Kind regards.

**Comment by scottwestover** [comment]
Hi @timu , you can have the widget hidden at the start of the chat by having the `showOnInit` configuration set to `false`. This setting is set in the configuration that was added to your account for the widget.

From there, you can send a notification when a specific event happens, and this will notify the user to open the widget window.

Unfortunately, I am not aware of any way to automatically open/close the window based on an event.

@liveperson Is anyone aware of a way to do this?

**Comment by orenkatz** [comment]
If you want to close the window, you can use the dispose command but keep in mind that you can't reopen it after you closed it.

What you can do is to use html & CSS and to hide the content at the beginning of the chat and show it only when a specific phrase is written in the chat.

**Comment by shaymLP** [comment]
@timu As it seems like your question has been answered, this issue will be closed now. If you have any further questions please refer to our new look [Support Forum](https://support.liveperson.com/forum/).
